kivy/core/text/text_layout.pyx

```diff
@@ -565,6 +565,7 @@ def layout_text(object text, list lines, tuple size, tuple text_size,
     # above, we might have gone a few lines over
     if max_lines > 0 and len(lines) > max_lines:
         val = True
+        h-= sum(l.h for l in lines[max_lines:])
```

Comment:
Sorry, I was just giving you a starting point - this is still not correct. Depending on the condition value below `dwn`, different lines are removed. See below which lines are deleted and they need to be subtracted.

Reply:
Did it. Thank you.